git-publish

```diff
@@ -510,12 +510,17 @@ def inspect_menu(tmpdir, to_list, cc_list, patches, suppress_cc, in_reply_to,
         elif a == 'e':
             edit(*patches)
         elif a == 's':
-            listfile = tempfile.NamedTemporaryFile()
-            listfile.write("\n".join(patches).encode(ENCODING))
-            listfile.flush()
-            edit(listfile.name)
-            listfile.seek(0)
-            patches = [x for x in listfile.read().splitlines() if len(x.strip())]
+            fd, tmpfile = tempfile.mkstemp(text=True)
+            fd_opened = os.fdopen(fd, 'wb')
+            fd_opened.write(os.linesep.join(patches).encode(ENCODING))
+            fd_opened.close()
+
+            edit(tmpfile)
+            fd_opened = open(tmpfile, 'rb')
+            patches = [x for x in fd_opened.read().splitlines() if len(x.strip())]
+            fd_opened.close()
+            if tmpfile:
```

Comment:
Can tmpfile ever evaluate to false? I guess mkstemp() raises an exception on error so tmpfile is always a valid path.